website/src/components/docCarousel/index.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import 'swiper/css';
+import { Navigation } from 'swiper';
+import 'swiper/css/navigation';
+
+
+function DocCarousel({ children }) {
+    console.log(children)
+    return (
+        <div className='docswiper'>
+            <Swiper
+                spaceBetween={30}
+                slidesPerView={1}
+                effect="fade"
+                navigation
+                modules={[Navigation]}
+
+                breakpoints={{
+                    640: {
+                        slidesPerView: 2,
+                        spaceBetween: 30,
+                    },
+                }}
+            >
+                {children.map((item) => <SwiperSlide>{item}</SwiperSlide>)}
+            </Swiper>
+        </div>
+    );
```

Comment:
For this `children.map`, one potential scenario that could cause an error is if the DocCarousel component is added to a page but does not wrap any content, which would make `children` undefined.

![image](https://user-images.githubusercontent.com/13647110/230932067-30beb4ab-1024-4809-9746-cc560e548cc9.png)

We should add a check here to ensure the carousel has `children` and that the `children` array length is greater than zero before adding any swiper html. There's a couple ways you can do this, but in my opinion, I would do this check before any of the swiper html (starting at `<div className='docswiper'>`) is rendered.

Before the return statement on line 10, you can check if `children` does not exist. If children does not exist return `null`, otherwise, continue on to return the swiper html.

Let me know if you have any questions with this. The [cta component](https://github.com/dbt-labs/docs.getdbt.com/blob/71f9f0efc824152b7d9956c940513b2f3a7a4a07/website/src/components/cta/index.js#L11) in this repo has a good example on doing a check before rendering content.